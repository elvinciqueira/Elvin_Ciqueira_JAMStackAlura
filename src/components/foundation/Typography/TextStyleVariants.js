import {css} from 'styled-components'

const paragraph1 = css`
  ${({theme}) => css`
    font-size: ${theme.typographyVariants.paragraph1.fontSize};
    font-weight: ${theme.typographyVariants.paragraph1.fontWeight};
    font-family: ${theme.typographyVariants.paragraph1.fontFamily};
  `}
`

const title = css`
  ${({theme}) => css`
    font-size: ${theme.typographyVariants.title.fontSize};
    font-weight: ${theme.typographyVariants.title.fontWeight};
    font-family: ${theme.typographyVariants.title.fontFamily};
    text-transform: ${theme.typographyVariants.title.textTransform};
  `}
`

const subTitle = css`
  ${({theme}) => css`
    font-size: ${theme.typographyVariants.subTitle.fontSize};
    font-weight: ${theme.typographyVariants.subTitle.fontWeight};
    font-family: ${theme.typographyVariants.subTitle.fontFamily};
  `}
`

const bodyText = css`
  ${({theme}) => css`
    font-size: ${theme.typographyVariants.bodyText.fontSize};
    font-weight: ${theme.typographyVariants.bodyText.fontWeight};
    font-family: ${theme.typographyVariants.bodyText.fontFamily};
  `}
`

const subTitle2 = css`
  ${({theme}) => css`
    font-size: ${theme.typographyVariants.subTitle2.fontSize};
    font-weight: ${theme.typographyVariants.subTitle2.fontWeight};
    font-family: ${theme.typographyVariants.subTitle2.fontFamily};
  `}
`

const xs = css`
  ${({theme}) => css`
    font-size: ${theme.typographyVariants.xs.fontSize};
    font-weight: ${theme.typographyVariants.xs.fontWeight};
    font-family: ${theme.typographyVariants.xs.fontFamily};
  `}
`

export const TextStyleVariants = {
  paragraph1,
  title,
  subTitle,
  subTitle2,
  xs,
  bodyText,
}
