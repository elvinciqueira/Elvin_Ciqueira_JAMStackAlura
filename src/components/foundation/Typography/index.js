import React from 'react'
import styled, {css} from 'styled-components'
import PropTypes from 'prop-types'
import get from 'lodash/get'
import {Link} from '../../common/Link'
import {propToStyle} from '../../../theme/utils/propToStyle'

const paragraph1 = css`
  ${({theme}) => css`
    font-size: ${theme.typographyVariants.paragraph1.fontSize};
    font-weight: ${theme.typographyVariants.paragraph1.fontWeight};
    font-family: ${theme.typographyVariants.paragraph1.fontFamily};
  `}
`

const title = css`
  ${({theme}) => css`
    font-size: ${theme.typographyVariants.title.fontSize};
    font-weight: ${theme.typographyVariants.title.fontWeight};
    font-family: ${theme.typographyVariants.title.fontFamily};
    text-transform: ${theme.typographyVariants.title.textTransform};
  `}
`

const subTitle = css`
  ${({theme}) => css`
    font-size: ${theme.typographyVariants.subTitle.fontSize};
    font-weight: ${theme.typographyVariants.subTitle.fontWeight};
    font-family: ${theme.typographyVariants.subTitle.fontFamily};
  `}
`

const subTitle2 = css`
  ${({theme}) => css`
    font-size: ${theme.typographyVariants.subTitle2.fontSize};
    font-weight: ${theme.typographyVariants.subTitle2.fontWeight};
    font-family: ${theme.typographyVariants.subTitle2.fontFamily};
  `}
`

const xs = css`
  ${({theme}) => css`
    font-size: ${theme.typographyVariants.xs.fontSize};
    font-weight: ${theme.typographyVariants.xs.fontWeight};
    font-family: ${theme.typographyVariants.xs.fontFamily};
  `}
`

export const TextStyleVariants = {
  paragraph1,
  title,
  subTitle,
  subTitle2,
  xs,
}

const TextBase = styled.span`
  ${({variant}) => TextStyleVariants[variant]}
  color: ${({theme, color}) => get(theme, `colors.${color}.color`)};
  ${propToStyle('textAlign')}
  ${propToStyle('margin')}
`

export default function Typography({tag, variant, children, href, ...rest}) {
  if (href) {
    return (
      <TextBase
        as={Link}
        variant={variant}
        href={href}
        // eslint-disable-next-line react/jsx-props-no-spreading
        {...rest}
      >
        {children}
      </TextBase>
    )
  }

  return (
    <TextBase as={tag} variant={variant} {...rest}>
      {children}
    </TextBase>
  )
}

Typography.defaultProps = {
  tag: 'span',
  variant: 'paragraph1',
}

Typography.propTypes = {
  children: PropTypes.node,
  tag: PropTypes.oneOf([
    'h1',
    'h2',
    'h3',
    'h4',
    'h5',
    'p',
    'li',
    'a',
    'span',
    'input',
    'textarea',
  ]),
  variant: PropTypes.oneOf([
    'paragraph1',
    'title',
    'XS',
    'subTitle',
    'subTitle2',
  ]),
}
