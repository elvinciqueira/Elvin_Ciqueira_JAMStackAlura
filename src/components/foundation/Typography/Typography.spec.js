/* eslint-disable no-undef */
import React from 'react'
import Typography from './index'

describe('<Typography />', () => {
  describe('when component renders', () => {
    test('it applies default styles', () => {
      const tree = rendererCreateWithTheme(<Typography />).toJSON()

      expect(tree).toMatchSnapshot()
      expect(tree).toHaveStyleRule('font-size', '1.75rem')
      expect(tree).toHaveStyleRule('font-weight', '400')
      expect(tree).toHaveStyleRule(
        'font-family',
        'Fira Sans Condensed,sans-serif',
      )
    })

    test('textAlign props sets text-align', () => {
      const tree = rendererCreateWithTheme(
        <Typography textAlign="center" />,
      ).toJSON()

      expect(tree).toMatchSnapshot()
      expect(tree).toHaveStyleRule('text-align', 'center')
    })

    test('margin props sets margin', () => {
      const tree = rendererCreateWithTheme(
        <Typography margin="20px" />,
      ).toJSON()

      expect(tree).toMatchSnapshot()
      expect(tree).toHaveStyleRule('margin', '20px')
    })

    test('fontSize props sets font-size', () => {
      const tree = rendererCreateWithTheme(
        <Typography fontSize="1rem" />,
      ).toJSON()

      expect(tree).toMatchSnapshot()
      expect(tree).toHaveStyleRule('font-size', '1rem')
    })
  })
})
