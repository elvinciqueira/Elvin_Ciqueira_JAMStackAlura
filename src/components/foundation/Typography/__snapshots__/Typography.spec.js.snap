// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Typography /> when component renders fontSize props sets font-size 1`] = `
.c0 {
  font-size: 1.75rem;
  font-weight: 400;
  font-family: Fira Sans Condensed,sans-serif;
  font-size: 1rem;
}

<span
  className="c0"
  fontSize="1rem"
/>
`;

exports[`<Typography /> when component renders it applies default styles 1`] = `
.c0 {
  font-size: 1.75rem;
  font-weight: 400;
  font-family: Fira Sans Condensed,sans-serif;
}

<span
  className="c0"
/>
`;

exports[`<Typography /> when component renders margin props sets margin 1`] = `
.c0 {
  font-size: 1.75rem;
  font-weight: 400;
  font-family: Fira Sans Condensed,sans-serif;
  margin: 20px;
}

<span
  className="c0"
/>
`;

exports[`<Typography /> when component renders textAlign props sets text-align 1`] = `
.c0 {
  font-size: 1.75rem;
  font-weight: 400;
  font-family: Fira Sans Condensed,sans-serif;
  text-align: center;
}

<span
  className="c0"
/>
`;
